bootstrap/kubeadm/internal/locking/control_plane_init_mutex.go

```diff
@@ -62,16 +62,25 @@ func (c *ControlPlaneInitMutex) Lock(ctx context.Context, cluster *clusterv1.Clu
 	case err != nil:
 		log.Error(err, "Failed to acquire lock")
 		return false
-	default: // successfully found an existing config map
+	default: // Successfully found an existing config map.
 		info, err := sema.information()
 		if err != nil {
 			log.Error(err, "Failed to get information about the existing lock")
 			return false
 		}
-		// the machine requesting the lock is the machine that created the lock, therefore the lock is acquired
+		// The machine requesting the lock is the machine that created the lock, therefore the lock is acquired.
 		if info.MachineName == machine.Name {
 			return true
 		}
+
+		// If the machine that created the lock can not be found unlock the mutex.
+		err = c.client.Get(ctx, client.ObjectKey{
+			Namespace: cluster.Namespace,
+			Name:      info.MachineName,
+		}, &clusterv1.Machine{})
+		if apierrors.IsNotFound(err) {
+			c.Unlock(ctx, cluster)
```

Comment:
The one issue here is whether or not Unlocking (even an invalid lock) is outside the remit of a Lock() function. In that case we could have validation of the lock somewhere else.

Reply:
I'm fine to have this here because, in the end, it is just an enhanced strategy to try to get the lock.
But let's wait for others opinions as well

Reply:
+1